tests/test_pvc_concurrent_deletion_creation.py

```diff
@@ -3,46 +3,89 @@
 """
 import os
 import subprocess
+import logging
 import pytest
 import yaml
-import logging
-from munch import munchify
+from utility.utils import run_cmd
+from utility.utils import delete_file
 import ocs.defaults as defaults
 
 from ocs import ocp
 from ocs import exceptions
+from ocs import volumes
 from ocsci import tier1, ManageTest
 from utility import templating
 
 log = logging.getLogger(__name__)
 TEMPLATE_DIR = "templates/ocs-deployment"
 TEMP_YAML_FILE_SC = '/tmp/temp_file_sc.yaml'
-TEMP_YAML_FILE_PVC = '/tmp/temp_file_pvc.yaml'
+TEST_PROJECT = 'test-project'
 SC = ocp.OCP(kind='StorageClass', namespace=defaults.ROOK_CLUSTER_NAMESPACE)
-PVC = ocp.OCP(kind='PersistentVolumeClaim',
-              namespace=defaults.ROOK_CLUSTER_NAMESPACE)
-PV = ocp.OCP(kind='PersistentVolume',
-             namespace=defaults.ROOK_CLUSTER_NAMESPACE)
+PVC = ocp.OCP(kind='PersistentVolumeClaim', namespace=TEST_PROJECT)
+PV = ocp.OCP(kind='PersistentVolume', namespace=TEST_PROJECT)
+PVOLC = volumes.PVC(namespace=TEST_PROJECT)
 
 
 @pytest.fixture(scope='class')
 def test_fixture(request):
     """
-    Removing the resources created for this test
+    Setup and teardown
     """
+    self = request.node.cls
+
     def finalizer():
-        teardown()
+        teardown(self)
     request.addfinalizer(finalizer)
+    setup(self)
+
+
+def setup(self):
+    """
+    Create project
+    Create storage class
+    Create PVCs
+    """
+    sc_parms = {'storageclass_name': 'test-sc', 'blockPool': 'rbd',
+                'k8s_api_version': defaults.STORAGE_API_VERSION}
+
+    # Create project
+    assert run_cmd(f'oc new-project {TEST_PROJECT}')
+
+    # Create storage class
+    create_storage_class(**sc_parms)
 
+    # Create 100 PVCs
+    create_multiple_pvc(self.pvc_base_name, self.number_of_pvc)
 
-def teardown():
+    # Verify PVCs are Bound
+    for count in range(1, self.number_of_pvc + 1):
+        pvc_name = f'{self.pvc_base_name}{count}'
+        pv = verify_pvc_and_fetch_pv_name(pvc_name)
+        assert pv, f'PVC {pvc_name} does not exists'
+        self.initial_pvs.append(pv)
+
+
+def teardown(self):
     """
+    Delete PVCs
     Delete storage class
     Delete temporary yaml files
+    Delete project
     """
+    # Delete newly created PVCs
+    command = (f'for i in `seq 1 {self.number_of_pvc}`;do oc delete pvc '
+               f'{self.pvc_base_name_new}$i -n {TEST_PROJECT};done')
+    proc = run_async(command)
+    assert proc, (f'Failed to execute command for deleting '
+                  f'{self.number_of_pvc} PVCs')
+
+    # Verify command to delete PVCs
+    ret, _, _ = proc.async_communicate()
+    assert not ret, "Deletion of newly created PVCs failed"
+
     assert SC.delete(yaml_file=TEMP_YAML_FILE_SC)
-    os.remove(TEMP_YAML_FILE_PVC)
-    os.remove(TEMP_YAML_FILE_SC)
+    assert run_cmd(f"oc delete project {TEST_PROJECT}")
```

Comment:
1) you can use ocp.py delete()
2) I think you first need to move to a different project

Reply:
Done